### ISSUE TITLE
Detect stability attributes on modules and crates

### ISSUE THREAD

**Issue by huonw** [rust-lang/rust#8962]
```rust
// some_crate.rs
#[experimental];

#[deprecated]
pub mod some_mod {
    pub fn foo() {}
} 
```

```rust
extern mod some_crate; // warning: use of experimental item

use some_crate::some_mod; // warning: use of deprecated item

fn main() {
    some_crate::some_mod::foo(); // warning: use of deprecated item
}
```

It seems unlikely that this should flag crates and modules without annotations, but I'm not really sure.

(Part of #6875, continuation of #8921.)
